### PR TITLE
Show appropriate error message when extension gallery service is not accessible

### LINKED ISSUE #55723: Show appropriate error message when extension gallery service is not accessible
Issue Type: <b>Bug</b>

Clicking the "Extensions" sidebar button shows nothing.  Searching for anything yields no results either.  (And the interface to show you that is pretty shoddy, doesn't really give you any text or indication that anything happened other than some small blue underline moving thing that is pretty ambiguous.)

https://i.imgur.com/9zyrk2s.png

I watched a YouTube video of someone's fresh install and it shows a giant list of Extensions so... not sure what is going on.

Thanks.

PS:  You get an error 500 from Github if you don't login first to submit this.

VS Code version: Code 1.25.1 (1dfc5e557209371715f655691b1235b6b26a06be, 2018-07-11T15:43:53.668Z)
OS version: Windows_NT x64 6.1.7601

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5 CPU         750  @ 2.67GHz (4 x 2664)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: disabled_software<br>rasterization: unavailable_software<br>video_decode: enabled<br>video_encode: enabled<br>vpx_decode: unavailable_software<br>webgl: enabled<br>webgl2: enabled|
|Memory (System)|15.99GB (11.83GB free)|
|Process Argv|C:\Program Files\Microsoft VS Code\Code.exe|
|Screen Reader|no|
|VM|0%|

</details>Extensions: none

### PR DESCRIPTION
**Pull request by aps32221**
### What does this PR do?

Update the `query` method to return the error thrown by the `this.extensionsWorkbenchService.queryGallery`

### What issues does this PR fix or reference?

Microsoft/vscode#55723

### CODE REVIEW

File: src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts [hunk 1]
```diff
@@ -240,7 +240,8 @@ export class ExtensionsListView extends ViewletPanel {
 
 		if (names.length) {
 			return this.extensionsWorkbenchService.queryGallery({ names, source: 'queryById' })
-				.then(pager => new PagedModel(pager));
+				.then(pager => new PagedModel(pager),
```
Comment (ramya-rao-a) [inline src/vs/workbench/parts/extensions/electron-browser/extensionsViews.ts]: Are you sure this is the part where the error is not bubbling up? The absence of error callback in a `then` would mean that the rejected promise would bubble up. I believe the problem is when we await of the `queryGalley` calls which occurs quite a few times in the `query` method.

A few ways to fix this
- Add try/catch around all the awaits on queryGallery 
- Convert the await to normal promise. So 
```
const pager = await this.extensionsWorkbenchService.queryGallery(options);
		return new PagedModel(pager);
```
would become
```
return this.extensionsWorkbenchService.queryGallery(options).then(pager => new PagedModel(pager));
```

**Review by ramya-rao-a — CHANGES_REQUESTED** [review]
Are you sure this is the part where the error is not bubbling up? The absence of error callback in a `then` would mean that the rejected promise would bubble up. I believe the problem is when we await of the `queryGalley` calls which occurs quite a few times in the `query` method.

A few ways to fix this
- Add try/catch around all the awaits on queryGallery 
- Convert the await to normal promise. So 
```
const pager = await this.extensionsWorkbenchService.queryGallery(options);
		return new PagedModel(pager);
```
would become
```
return this.extensionsWorkbenchService.queryGallery(options).then(pager => new PagedModel(pager));
```

**Comment by aps32221** [comment]
@ramya-rao-a thanks for the review.
I think I got the wrong point for this problem.
And you are right. Thank you to point that.
```
const pager = await this.extensionsWorkbenchService.queryGallery(options);
		return new PagedModel(pager);
```
This code always returned a valid paged model.
So convert all await to normal promise in `query` when we await `queryGallery`.
Am I right?
I will fix it as soon as I can.
BTW my English is not very good. if something I got wrong, plz tell me, thank!

**Comment by ramya-rao-a** [comment]
> So convert all await to normal promise in query when we await queryGallery.
Am I right?

Yes

**Comment by ramya-rao-a** [comment]
@aps32221 There was another PR #56767 that fixed the issue, so I'll be closing this PR.

Feel free to look for other issues that are marked as "help-wanted" to contribute.

Thanks again

**Pull request closed without merging**